FiducialRegistrationWizard/Logic/vtkPointMatcher.cxx

```diff
@@ -0,0 +1,368 @@
+#include "vtkPointMatcher.h"
+#include "vtkCombinatoricGenerator.h"
+#include <vtkMath.h>
+
+#define RESET_VALUE_COMPUTED_ROOT_MEAN_DISTANCE_ERROR VTK_DOUBLE_MAX
+#define MINIMUM_NUMBER_OF_POINTS_NEEDED_TO_MATCH 3
+
+//----------------------------------------------------------------------------
+vtkStandardNewMacro( vtkPointMatcher );
+
+//------------------------------------------------------------------------------
+vtkPointMatcher::vtkPointMatcher()
+{
+  // point lists, even outputs, should never be null
+  this->InputPointList1 = vtkSmartPointer< vtkPoints >::New();
+  this->InputPointList2 = vtkSmartPointer< vtkPoints >::New();
+  this->MaximumDifferenceInNumberOfPoints = 2;
+  this->TolerableRootMeanSquareDistanceErrorMm = 50.0;
+  this->ComputedRootMeanSquareDistanceErrorMm = RESET_VALUE_COMPUTED_ROOT_MEAN_DISTANCE_ERROR;
+  this->OutputPointList1 = vtkSmartPointer< vtkPoints >::New();
+  this->OutputPointList2 = vtkSmartPointer< vtkPoints >::New();
+
+  // timestamps for input and output are the same, initially
+  this->InputChangedTime.Modified();
+  this->OutputChangedTime.Modified();
+}
+
+//------------------------------------------------------------------------------
+vtkPointMatcher::~vtkPointMatcher()
+{
+}
+
+//------------------------------------------------------------------------------
+void vtkPointMatcher::PrintSelf( std::ostream &os, vtkIndent indent )
+{
+  Superclass::PrintSelf( os, indent );
+  
+  os << indent << "MaximumDifferenceInNumberOfPoints: " << this->MaximumDifferenceInNumberOfPoints << std::endl;
+  os << indent << "TolerableRootMeanSquareDistanceErrorMm: " << this->TolerableRootMeanSquareDistanceErrorMm << std::endl;
+  os << indent << "ComputedRootMeanSquareDistanceErrorMm: " << this->ComputedRootMeanSquareDistanceErrorMm << std::endl;
+  os << indent << "IsMatchingWithinTolerance: " << this->IsMatchingWithinTolerance() << std::endl;
+  os << indent << "UpdateNeeded: " << this->UpdateNeeded() << std::endl;
+}
+
+//------------------------------------------------------------------------------
+// INPUT MUTATORS
+//------------------------------------------------------------------------------
+
+//------------------------------------------------------------------------------
+void vtkPointMatcher::SetInputPointList1( vtkPoints* points )
```

Comment:
In general, in VTK classes Set(Object) functions do not make a deep copy.
See for example in vtkRenderView:

header

    virtual void SetTransform(vtkAbstractTransform* transform);
    vtkGetObjectMacro(Transform, vtkAbstractTransform);

implementation

    vtkCxxSetObjectMacro(vtkRenderView, Transform, vtkAbstractTransform);

You can use vtkSetObjectBodyMacro if you need to do some custom operations in addition to just storing a reference to the object.